docker-app/qfieldcloud/filestorage/migrations/0002_file_file_storage.py

```diff
@@ -0,0 +1,59 @@
+# Generated by Django 4.2.19 on 2025-05-06 06:48
+
+from django.db import connection, migrations, models
+from django.db.models import Count
+
+import qfieldcloud.core.models
+import qfieldcloud.core.validators
+
+
+def set_file_storage_for_files(apps, schema_editor):
+    # update File's file_storage value to the parent Project one.
+    with connection.cursor() as cursor:
+        cursor.execute(
+            """
+            UPDATE filestorage_file ff
+            SET
+                file_storage = cp.file_storage
+            FROM core_project cp
+                WHERE cp.id = ff.project_id
+        """
+        )
```

Comment:
I tried in pure python, but it creates either a query that does not run:
```
File.objects.filter(id=-9).update(name=F('project__file_storage'))
```
with `FieldError: Joined field references are not permitted in this query`

or supper inefficient query:
```
File.objects.update(
    file_storage=Subquery(Project.objects.filter(id=OuterRef('project_id')).values('file_storage')[:1])
)
```

So it seems we have to live with the raw SQL .